python/ray/tune/syncer.py

```diff
@@ -787,6 +791,9 @@ def _remote_trial_logdir(self, trial: "Trial"):
     def _sync_trial_dir(
         self, trial: "Trial", force: bool = False, wait: bool = True
     ) -> bool:
+        if not os.environ.get(REENABLE_DEPRECATED_SYNC_TO_HEAD_NODE):
+            return False
```

Comment:
Do we need the same error message here?

Reply:
Oh I see, this is for artifact syncing?

Reply:
If a user is running multi-node without checkpointing, but still saves artifacts:

- `on_checkpoint` does not get called
- Trial artifacts would no longer be pulled to the head node.

We can do one of these options:
1. Detect if the trial wrote any artifacts. If so, raise an error on this callback.
2. Just log a warning on the first time sync_trial_dir gets called for a remote trial and does a no-op, saying something like "Syncing to head node is disabled. If you are writing any trial artifacts, they will not be persisted. Please do X or Y." Logging on every sync_trial_dir will spam the logs too much.

I think option 2 is safer.